enhance: #49 Add button inside image url input to generate random images

jjinendra3 opened the pull request:
## Summary
Updated image modal to allow users to select any of the 7 images from the webp format instead of randomizing.

## Description
In this pull request, I've modified the image modal functionality to enhance user experience. Now, users can specifically choose from any of the 7 available images in the webp format instead of the previous randomization. This change was implemented to give users more control over the visuals they want to interact with. I have also fixed the code according to the feedback received in the previous PR.

## Images
https://drive.google.com/drive/folders/14JFGZFLu-vtUvZdi7CmX4zrihCY0Lv21

Please find the new video in the drive (Updated from the last PR)


## Issue(s) Addressed
Closes #49 

## Prerequisites
- [x] Have followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -6,6 +6,15 @@ import 'react-toastify/dist/ReactToastify.css';
+  'https://i.ibb.co/KLwfZzG/Maldives-1024x767.webp',
+  'https://i.ibb.co/qxFMj1H/sunset-horizon-clean-sky-nature.webp',
+];
 type FormData = {
```
Kindly move this to constants folder

jjinendra3 replied:
Done :)

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -58,11 +74,11 @@ function AddBlog() {
+    //   toast.error('Image URL must end with .jpg, .jpeg, or .png');
+    //   return false;
+    // }
     if (formData.categories.length > 3) {
```
Why are we commenting this ? can we instead add WebP in this regex

jjinendra3 replied:
Added :)

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -182,6 +211,79 @@ function AddBlog() {
         </button>
       </form>
+      {modal && ( //MODAL STARTS
+        <div className="relative z-10"
```
Please create a new component and use it here. Helps us in maintaining the code in long run , see which state management helps, for now you can also pass state down the prop, if you feel it very challenging to setup the state by yourself, you can pass handleImageSelect and setFormData in the Modal component and set it there, you can explore little on this part, the good practice would be context api for now if you are willing to explore, else passing the props should do :)

jjinendra3 replied:
I'm relatively new to TypeScript and haven't worked with TypeScript + React before. As a result, I couldn't complete the task when you mentioned earlier. Could we open a new refactoring issue to address this after the mege, please lmk your thoughts on this.

krishnaacharyaa reviewed (changes requested):
Thank you @jjinendra3, Kindly address the comments

krishnaacharyaa commented:
Deployment seems to fail because of the incompatible types, kindly try to fix that, let me know you need any help fixing that, happy to help :)

jjinendra3 commented:
> Deployment seems to fail because of the incompatible types, kindly try to fix that, let me know you need any help fixing that, happy to help :)

![image](https://github.com/krishnaacharyaa/wanderlust/assets/117433354/ea2af589-1459-42a3-9cae-5cc1e5146ab2)

 It seems the issue was arising from the `setFormData` call, where the `selectedImage` variable had a type mismatch. Initially, the `useState` for selecting images was set to `null`, causing a compatibility issue since `selectedImage` was not strictly a string.

I've addressed this by introducing a conditional check to ensure `selectedImage` is a string before updating the state as you can see above.This modification should resolve the type incompatibility, and I believe deploying the updated code should no longer pose issues.

Regarding the reviews you requested, all have been addressed except for the modal part. Managing props in TypeScript is different from JavaScript. I suggest creating a new refactoring issue for the same.

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -15,6 +16,13 @@ type FormData = {
 };
 function AddBlog() {
+  const [selectedImage, setSelectedImage] = useState<string | null>("");
+
```
Why is null required here ? "" would be by default, what is the case where the selected image is null ?

jjinendra3 replied:
Fixed

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -57,8 +75,7 @@ function AddBlog() {
+    const imageLinkRegex = /\.(jpg|jpeg|png|webp)$/i;
     if (!imageLinkRegex.test(formData.imageLink)) {
       toast.error('Image URL must end with .jpg, .jpeg, or .png');
       return false;
```
Please add .webp even in the toast message

jjinendra3 replied:
Added

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -122,15 +139,27 @@ function AddBlog() {
+          <button
+            type="button"
+            className="bg-black text-white items-center justify-center text-base p-2 rounded-lg hover:bg-gray-800 w-full md:w-fit flex flex-2"
+            onClick={() => {
```
Here can we have it's height same to the image url, don't mind I'm a little picky; because according to the UX, this button out shines more than the create blog button.

jjinendra3 replied:
Fixed

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -42,7 +50,17 @@ function AddBlog() {
+  const handleselector = () => {
+    var url = "";
+    if (typeof selectedImage === 'string') {
+      url = selectedImage;
```
Removing null should help us removing this overhead condition

jjinendra3 replied:
Done, removed

krishnaacharyaa reviewed (changes requested):
Thank you for addressing the issues, kindly address these small changes, And about the props passing I get it state may be a bit tricky, but other than js this just involves type in it, shouldn't be as tough as you tought, will create new issue anyways, no issues :)

jjinendra3 commented:
> Thank you for addressing the issues, kindly address these small changes, And about the props passing I get it state may be a bit tricky, but other than js this just involves type in it, shouldn't be as tough as you tought, will create new issue anyways, no issues :)

I reasearched on it and finally found the way  to do it :) So the refactoring is done, no need to create a new issue for the same

krishnaacharyaa reviewed (approved):
Thank you! Looks good to me ☺️
